flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImpl.java

```diff
@@ -129,7 +129,7 @@ public void addOutputData(long checkpointId, ResultSubpartitionInfo info, int st
 			info,
 			startSeqNum,
 			data == null ? 0 : data.length);
-		enqueue(write(checkpointId, info, checkBufferType(data)), false);
+		enqueue(write(checkpointId, info, data), false);
```

Comment:
Why have you dropped `checkBufferType` call? It was actually quite helpful couple of times when debugging. Can not we try to keep it?

Reply:
It is now called when building `iterator` - in the same way as for `InputChannel` buffers.